src/Model/ModelManager.php

```diff
@@ -77,11 +77,7 @@ public function __construct(ManagerRegistry $registry, ?PropertyAccessorInterfac
      */
     public function getMetadata($class)
     {
-        $metadata = $this->getDocumentManager($class)->getMetadataFactory()->getMetadataFor($class);
-
-        \assert($metadata instanceof MongoDBClassMetadata);
-
-        return $metadata;
+        return $this->getDocumentManager($class)->getClassMetadata($class);
```

Comment:
I have trouble to consider this change as pedantic 

You change
```
->getMetadataFactory()->getMetadataFor($class);
```
to 
```
->getClassMetadata($class);
```

Reply:
Pedantic as a refactor the user won't notice.